lib/utils.pm

```diff
@@ -1663,6 +1664,54 @@ sub script_output_retry {
     die("Waiting for Godot: $cmd") if $die;
 }
 
+
+=head2 validate_script_output_retry
+
+ validate_script_output_retry($cmd, $check, [retry => $retry], [delay => $delay], [timeout => $timeout]);
+
+Repeat command until validate_script_output succeeds or die. Return the output of the command on success.
+
+C<$retry> refers to the number of retries and defaults to C<10>.
+
+C<$delay> is the time between retries and defaults to C<30>.
+
+If the command doesn't succeed after C<$retry> retries,
+this function will die.
+
+Example:
+
+ validate_script_output_retry('ping -c1 -W1 machine', m/1 packets transmitted/, retry => 5, delay => 60);
+
+=cut
+sub validate_script_output_retry {
+    my ($cmd, $check, %args) = @_;
+    $args{retry} //= 10;
+    $args{delay} //= 30;
+    $args{timeout} //= 90;
+    $args{proceed_on_failure} //= 1;
+    my $retry = delete $args{retry};
+    my $delay = delete $args{delay};
+    my $ret;
+
+    my $exec = "timeout $args{timeout} $cmd";
+    # Exclamation mark needs to be moved before the timeout command, if present
+    if (substr($cmd, 0, 1) eq "!") {
+        $cmd = substr($cmd, 1);
+        $cmd =~ s/^\s+//;    # left trim spaces after the exclamation mark
+        $exec = "! timeout $args{timeout} $cmd";
+    }
+
+    $args{timeout} += 3;    # timeout for script_run must be larger than for the 'timeout ...' command
+    for (1 .. $retry) {
+        eval { $ret = validate_script_output($exec, $check, %args); };
```

Comment:
```suggestion
        my $timeout = delete $args{timeout};
        eval { $ret = validate_script_output($exec, $check, $timeout, %args); };
```
I think this is the workaround for the failing unit-test. It's because of https://github.com/os-autoinst/os-autoinst/pull/1974 and you are passing a hash, where the order of the elements is random. 
As you set `proceed_on_failure` in that hash and this key isn't handled properly in `validate_script_output` you sometimes fail...